sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -139,17 +139,19 @@ class MapboxMap :
    *
    * @param styleUri The style URI
    * @param styleTransitionOptions style transition options applied when loading the style
+   * @param onStylePreLoaded The OnStylePreloaded callback
```

Comment:
Imho might be dangerous. Users can (and some will definitely, I suppose) accidentally use `onStylePreLoaded` and treat style as loaded, using APIs not allowed yet

Reply:
There's no APIs that's not allowed, it's just some use cases, like modifying a source property that has not been loaded will cause some issues.

Reply:
And these dangerous use cases is clearly documented in the `onStylePreLoaded` listener, so users should use this API at their own risk.

Reply:
I mean - it's easy to mistakenly use `map.loadStyle(onStylePreloaded = {})` instead of `map.loadStyle(onStyleLoaded = {})`, calls look similarly enough too confuse them writing manually or with autocomplete

Reply:
> There's no APIs that's not allowed, it's just some use cases, like modifying a source property that has not been loaded will cause some issues.

IMHO it's close to `not allowed`, if the api call can cause issues, even if this doesn't mean crash =)

Reply:
> I mean - it's easy to mistakenly use map.loadStyle(onStylePreloaded = {}) instead of map.loadStyle(onStyleLoaded = {}), calls look similarly enough too confuse them writing manually or with autocomplete

Do you have suggestion on the naming so that it will become less confusing?

> IMHO it's close to not allowed, if the api call can cause issues, even if this doesn't mean crash =)

The misuse will most likely cause errors like `Source with Source id is not found in the style` I assume 😄